src/dumbo/src/ns.rs

```diff
@@ -360,10 +369,21 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_ns_new_with_defaults() {
+        let ns = MmdsNetworkStack::new_with_defaults(None);
+        assert_eq!(ns.mac_addr, MacAddr::parse_str(DEFAULT_MAC_ADDR).unwrap());
+        assert_eq!(ns.ipv4_addr, Ipv4Addr::from(DEFAULT_IPV4_ADDR));
+
+        let ns = MmdsNetworkStack::new_with_defaults(Some(Ipv4Addr::LOCALHOST));
+        assert_eq!(ns.mac_addr, MacAddr::parse_str(DEFAULT_MAC_ADDR).unwrap());
+        assert_eq!(ns.ipv4_addr, Ipv4Addr::LOCALHOST);
+    }
+
     #[test]
     #[allow(clippy::cognitive_complexity)]
     fn test_ns() {
-        let mut ns = MmdsNetworkStack::new_with_defaults();
+        let mut ns = MmdsNetworkStack::new_with_defaults(Some(Ipv4Addr::from(DEFAULT_IPV4_ADDR)));
         assert_eq!(ns.mac_addr, MacAddr::parse_str(DEFAULT_MAC_ADDR).unwrap());
         assert_eq!(ns.ipv4_addr, Ipv4Addr::from(DEFAULT_IPV4_ADDR));
```

Comment:
> Are these asserts useful anymore? It looks like you are already checking these in the previous test.

They are not needed. Removed them.

>Also, I think new_with_defaults() should be called here with None parameter as calling it with the default value seems like a very unlikely scenario, what do you think?

I've used this way of initializing the `MmdsNetworkStack` because I wanted to make obvious what IPv4 address is used, to be more readable. It might be simpler instead if the code is less polluted. Simplified by calling `new_with_defaults(None)`, as you suggested. Removed also the asserts which unit test the `new_with_default` function, actually.